lib/screens/explore_screens/vote_screens/proposal_details/interactor/usecases/get_proposal_data_use_case.dart

```diff
@@ -4,40 +4,59 @@ import 'package:seeds/datasource/local/settings_storage.dart';
 import 'package:seeds/datasource/remote/api/profile_repository.dart';
 import 'package:seeds/datasource/remote/api/proposals_repository.dart';
 import 'package:seeds/datasource/remote/api/voice_repository.dart';
-import 'package:seeds/datasource/remote/model/proposal_model.dart';
 import 'package:seeds/datasource/remote/model/vote_model.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/proposals/viewmodels/proposal_view_model.dart';
 
 class GetProposalDataUseCase {
   final ProfileRepository _profileRepository = ProfileRepository();
   final ProposalsRepository _proposalsRepository = ProposalsRepository();
   final VoiceRepository _voiceRepository = VoiceRepository();
 
-  Future<List<Result>> run(ProposalModel proposal) {
+  Future<List<Result>> run(ProposalViewModel proposal) {
     final userAccount = settingsStorage.accountName;
-    final _getVoice = proposal.campaignType == 'alliance'
-        ? _voiceRepository.getAllianceVoice(userAccount)
-        : _voiceRepository.getCampaignVoice(userAccount);
+    final _getVoice = _getVoiceCall(categoryTypeLabel: proposal.categoryTypeLabel, userAccount: userAccount);
 
     final futures = [
       _profileRepository.getProfile(proposal.creator),
-      _fetchVote(proposal.id, userAccount),
+      _fetchVote(proposal, userAccount),
       _getVoice,
     ];
     return Future.wait(futures);
   }
 
-  Future<Result> _fetchVote(int proposalId, String account) async {
-    final cacheRepository = const CacheRepository();
-    VoteModel? voteModel = await cacheRepository.getProposalVote(account, proposalId);
-    if (voteModel == null) {
-      final result = await _proposalsRepository.getVote(proposalId, account);
+  Future<Result> _fetchVote(ProposalViewModel proposal, String account) async {
+    late Result result;
+    VoteModel? voteModel;
+    if (proposal.categoryTypeLabel == 'referendum') {
+      result = await _proposalsRepository.getReferendumVote(proposal.id, account);
```

Comment:
I agree but in the next PR

Reply:
It would be good I guess... it's the nature of a cache you never "have to" do it... but if we access this every time we scroll then probably better. 

I've realized votes caching is much more complex than I thought thanks to the "eval" stage and people being able to change their votes in eval...